app/app.go

```diff
@@ -76,6 +78,8 @@ func appHandler(c *echo.Context) error {
 				log.Error(err)
 				failed = true
 			}
+			appsMutex.Lock()
+			defer appsMutex.Unlock()
```

Comment:
Nitpick - it may be clearer to put `appsMutex.Unlock()` after `envApps[env] = controllers` instead of using a `defer`. Also `defer` has some overhead.